controllers/profiles/ensurer.go

```diff
@@ -33,11 +33,22 @@ func newObjectEnsurer(client client.Client, logger *logr.Logger, creator objectC
 	}
 }
 
+// newConditionedObjectEnsurer see objectEnsurer
+func newConditionedObjectEnsurer(client client.Client, logger *logr.Logger, creator objectCreator, conditions ...objectCreationCondition) *objectEnsurer {
+	return &objectEnsurer{
+		client:     client,
+		logger:     logger,
+		creator:    creator,
+		conditions: conditions,
+	}
+}
+
 // objectEnsurer provides the engine for a ReconciliationState that needs to create or update a given Kubernetes object during the reconciliation cycle.
 type objectEnsurer struct {
-	client  client.Client
-	logger  *logr.Logger
-	creator objectCreator
+	client     client.Client
+	logger     *logr.Logger
+	creator    objectCreator
+	conditions []objectCreationCondition
```

Comment:
Instead, I think we should have ensurers per cluster. This way, we create the right `ensurer` once we start the reconciliation and avoid looping in these conditions. In a nutshell, we should use the builder to have the right structure based on the conditionals we want, and not during the reconciliation cycle.